controls_rewrite_folder/driver_interface_error_handling.cpp

```diff
@@ -0,0 +1,47 @@
+#include <iostream>
+#include <cmath>
+//everson shen and aleeza ali zar
+
+
+
+double potentiometerProcessing(double fractionIn){
+
+    if (fractionIn >= 0 && fractionIn <= 1){
+        return fractionIn;
+    }
+    else{
+        return 0;
+    }
+}
+
+bool b_DriverInterfaceError(double DI_V_AccelPedalPos1, double DI_V_AccelPedalPos2, double DI_FrontBrakePressure, double DI_RearBrakePressure, double DI_V_RawSteeringAngle){
+
+
+    double PedalPos1 = potentiometerProcessing(DI_V_AccelPedalPos1);
+    double PedalPos2 = potentiometerProcessing(DI_V_AccelPedalPos2);
+    double FrontBrakePressure = potentiometerProcessing(DI_FrontBrakePressure);
+    double RearBrakePressure = potentiometerProcessing(DI_RearBrakePressure);
+    double RawSteeringAngle = potentiometerProcessing(DI_V_RawSteeringAngle);
+
+    return ((PedalPos1 == 0) || (PedalPos2 == 0) || (FrontBrakePressure == 0) || (RearBrakePressure == 0) || (RawSteeringAngle == 0) || (std::abs(DI_V_AccelPedalPos1 - DI_V_AccelPedalPos2) > 0.1));
```

Comment:
This function will return an error if all doubles are 0, even if the sensors truly are 0.

This is where we will start to diverge from the simulink model in order to get our desired behaviour:

We should replace  `potentiometerProcessing` with a function `bool isInRange(value)` that returns true if it is between 0-1 and false otherwise. Then you can simplify the interface error function.